src/dbnode/storage/bootstrap/bootstrapper/readers.go

```diff
@@ -206,9 +206,10 @@ func newShardReaders(
 
 		openOpts := fs.DataReaderOpenOptions{
 			Identifier: fs.FileSetFileIdentifier{
-				Namespace:  ns.ID(),
-				Shard:      shard,
-				BlockStart: blockStart,
+				Namespace:   ns.ID(),
+				Shard:       shard,
+				BlockStart:  blockStart,
+				VolumeIndex: info.VolumeIndex,
```

Comment:
Small bug here where we weren't passing along the VolumeIndex which was resulting in "checkpoint not found" errors. Additional context starting from [here](https://chronosphereio.slack.com/archives/CQG4P1SP9/p1597152431050000) in Slack. Also added to [2860](https://app.clubhouse.io/chronosphere/story/2860/establish-peer-bootstrapping-time-is-sane-reasonable-for-certain-workloads-and-have-integration-tests) to get integration test coverage

Reply:
Nice catch; can we make it harder to miss setting this in future by removing the explicit struct init here and hiding it behind a `fs.NewFileSetFileIdentifier(namespace ident.ID,blockStart time.Time, shard uint32, volumeIndex int) FileSetFileIdentifier` constructor?